### PR TITLE
support for binary builds when app contains label provider=fabric8

### LINKED ISSUE #7: Support Binary Builds When label "provider=fabric8" is present

### LINKED ISSUE #10: Enhance Test Framework With A Sophisticated Use Case
See https://github.com/rht-labs/dynamic-pipeline-jenkins-plugin/pull/9#pullrequestreview-5468886

### PR DESCRIPTION
**Pull request by sherl0cks**
#### What does this PR do?
Dynamically generate the openshift s2i binary builds required by fabric8 in order to avoid putting cluster details like hostname in custom build image commands. Requested by @oybed 

#### Which tests illustrate how this code works?
See unit tests that end in fabric8

#### Is there a relevant Issue open for this?
resolves #7 
resolves #10 

#### Are there any other relevant resources that should be reviewed as well?
Nope

#### Who would you like to review this?
/cc @mdanter @oybed

### CODE REVIEW

File: src/main/java/com/rhc/dynamic/pipeline/ReleasePipelineScriptGenerator.java [hunk 1]
```diff
@@ -83,7 +83,12 @@ private static String generateBuildImageAndDeployToDevStage(Engagement engagemen
 		StringBuilder script = new StringBuilder();
 		script.append("\n  stage ('Build Image and Deploy to Dev') {\n");
 		Application app = EngagementDAO.getApplicationFromBuildProject(engagement, applicationName);
-		if (app.getBuildImageCommands() == null || app.getBuildImageCommands().isEmpty()) {
+		if ( app.getLabels().containsKey("provider") && app.getLabels().get("provider").equals("fabric8")){
+			script.append("    echo 'Found label \"provider=fabric8\", we're generating the s2i binary build commands'\n");
+			script.append(String.format("    sh 'oc login %s --token=$OPENSHIFT_API_TOKEN --insecure-skip-tls-verify'%n", EngagementDAO.getBuildCluster(engagement).getOpenshiftHostEnv()));
```
Comment (oybed) [inline src/main/java/com/rhc/dynamic/pipeline/ReleasePipelineScriptGenerator.java]: Is the use of the `OPENSHIFT_API_TOKEN` documented somewhere? I.e.: that it is expected to be a parameter passed in Jenkins? If not, would be good to maybe see some comments or a write-up in a README somewhere.

File: src/main/java/com/rhc/dynamic/pipeline/ReleasePipelineScriptGenerator.java [hunk 2]
```diff
@@ -83,7 +83,12 @@ private static String generateBuildImageAndDeployToDevStage(Engagement engagemen
...
+			script.append(String.format("    sh 'oc start-build %s --from-dir=. --follow -n %s'", applicationName, EngagementDAO.getBuildProject(engagement).getName()));
```
Comment (oybed) [inline src/main/java/com/rhc/dynamic/pipeline/ReleasePipelineScriptGenerator.java]: Just to confirm - if there are multiple apps running in parallel in the same project, we'd expect separate / individual pipelines for each app and hence don't need to account for multiple builds in parallel here, correct?

File: src/main/java/com/rhc/dynamic/pipeline/ReleasePipelineScriptGenerator.java (hunk 1 above)
Comment (sherl0cks) [inline src/main/java/com/rhc/dynamic/pipeline/ReleasePipelineScriptGenerator.java]: Good call out. I'll make a change in the README

**Comment by sherl0cks** [comment]
@oybed changes made on the docs. I've opened #10 to address your testing concern.

**Comment by sherl0cks** [comment]
New commit closes #10

**Comment by oybed** [comment]
Looks good - merging.

**Pull request merged**